pkg/capi/edit/ProviderConfig.vue

```diff
@@ -0,0 +1,234 @@
+      }
+      set(this.value.spec.configSecret, 'name', this.generateName(this.provider)); // Defines the name of the secret that will be created or adjusted based on the content of the spec.features and spec.variables.
+    },
+    generateName(name: string) {
+      return name ? `${ name }-credentials-${ randomStr(5).toLowerCase() }` : undefined;
+    },
+    async getNamespaces() {
+      const inStore = this.$store.getters['currentStore'](NAMESPACE);
+
+      return await this.$store.dispatch(`${ inStore }/findAll`, { type: NAMESPACE });
+    },
+
+    async saveOverride() {
+      if ( this.errors ) {
+        clear(this.errors);
+      }
+      try {
+        await this.value.save();
+
+        return this.done();
+      } catch (err) {
+        this.errors.push(err);
+      }
+    },
+    cancelCredential() {
+      if ( this.$refs.cruresource ) {
+        this.$refs.cruresource.emitOrRoute();
+      }
+    }
+  }
+});
+</script>
+
+<template>
+  <Loading v-if="loading" />
+  <CruResource
+    v-else
+    :mode="mode"
+    :validation-passed="true"
+    :resource="value"
+    :errors="errors"
+    :cancel-event="true"
+    :done-route="doneRoute"
+    :apply-hooks="applyHooks"
+    class="provider"
+    component-testid="capi-provider-create"
+    @done="done"
+    @finish="saveOverride"
+    @cancel="done"
+    @error="fvUnreportedValidationErrors"
+  >
+    <NameNsDescription
+      v-if="!isView"
+      v-model="value"
+      :mode="mode"
+      :namespaced="true"
+      :namespace-options="allNamespaces"
+      name-label="capi.provider.name.label"
+      name-placeholder="capi.provider.name.placeholder"
+      description-label="capi.provider.description.label"
+      description-placeholder="capi.provider.description.placeholder"
+      :rules="{name:fvGetAndReportPathRules('metadata.name')}"
+    />
+    <div v-if="provider === 'custom'" class="col span-3 mb-40">
+      <LabeledInput
+        v-model="value.spec.name"
+        :mode="mode"
+        label-key="capi.provider.type.label"
+        required
+        :rules="fvGetAndReportPathRules('spec.name')"
+      />
+    </div>
+    <div v-if="needCredential" class="mb-40" />
+    <h2 class="mb-20">
+      <t k="capi.provider.secret.title" />
+    </h2>
+    <RadioGroup
+      v-model="shouldCreateSecret"
+      name="shouldCreateSecret"
+      :mode="mode"
+      :options="modeOptions"
+      class="mb-40"
+    />
+    <div v-if="shouldCreateSecret">
+      <SelectCredential
+        v-if="needCredential"
+        v-model="value.spec.credentials.rancherCloudCredential"
+        :mode="mode"
+        :provider="provider"
+        :cancel="cancelCredential"
+        :showing-form="showForm"
+        class="mb-40"
+      />
+      <h3 class="mb-20">
+        <t k="capi.provider.features.title" />
+      </h3>
+      <div class="mb-40">
+        <Checkbox
+          v-model="value.spec.features.clusterResourceSet"
```

Comment:
The default providers installed w/ turtles don't have `spec.features` defined so these checkboxes throw an error when editing them. There are different ways of handling that; I think the fastest would be to set `features: {}` when initializing the component but the more correct way is probably to change the value prop on checkboxes to something that wont break if spec.features is undefined and update the  input event handler to set spec.features if not defined

Reply:
I think we will need to discuss how to handle non-standard providers in general tomorrow. I will just take out that options until it is decided (hopefully, tomorrow).